src/Interceptor/DecompressResponse.php

```diff
@@ -61,8 +63,11 @@ private function decompressResponse(Response $response): Response
             $sizeLimit = $response->getRequest()->getBodySizeLimit();
             $decompressedBody = new ZlibInputStream($response->getBody(), $encoding);
 
+            $response->getRequest()->setAttribute(self::CONTENT_ENCODING_ATTR, $response->getHeader('content-encoding'));
             $response->setBody(new SizeLimitingInputStream($decompressedBody, $sizeLimit));
             $response->removeHeader('content-encoding');
+        } elseif ($response->getRequest()->hasAttribute(self::CONTENT_ENCODING_ATTR)) {
+            $response->getRequest()->removeAttribute(self::CONTENT_ENCODING_ATTR);
```

Comment:
This should really happen, no?

Reply:
I have in mind multiple redirects with alternatively compressed/uncompressed bodies. Makes sense?

Reply:
These shouldn't reuse the same request object, but clone it or use a new request object.

Reply:
OK, removed.